src/gm3/components/map/index.js

```diff
@@ -1062,67 +1007,110 @@ class Map extends React.Component {
+                    this.drawTool = new olSelectInteraction({
+                        layers: [this.olLayers[map_source_name]],
+                        style: null,
+                    });
+
+                    this.drawTool.on('select', evt => {
+                        modifyNext(
+                            [GEOJSON_FORMAT.writeFeatureObject(evt.selected[0])]
+                        );
+                    });
+                } else if (layer && layer.queryAs.length > 0) {
+                    // TODO: This wildly assumes the query-as layer is WFS,
+                    //       at the time of writing WFS was the only option.
+                    const queryLayer = mapSourceActions.getLayerFromPath(
+                        this.props.mapSources,
+                        layer.queryAs[0]
                     );
-                    // clear the selected features from the tool.
-                    this.drawTool.getFeatures().clear();
-                });
-            } else if(type === 'Modify') {
-                const features = source.getFeatures();
-                this.drawTool = new olModifyInteraction({
-                    features: new olCollection(features),
-                });
 
-                if(is_selection) {
-                    this.drawTool.on('modifyend', (evt) => {
-                        const features = evt.features.getArray();
-                        this.addSelectionFeatures(features, this.props.mapView.selectionBuffer);
+                    const editSrc = this.olLayers[EDIT_LAYER_NAME].getSource();
+
+                    this.drawTool = new olDrawInteraction({
+                        type: 'Point',
+                        source: editSrc,
                     });
-                } else {
-                    this.drawTool.on('modifyend', (evt) => {
-                        const id_prop = '_uuid';
-                        const features = evt.features.getArray();
-                        for(const feature of features) {
-                            const geometry = util.geomToJson(feature.getGeometry());
-                            const id = feature.getProperties()[id_prop];
 
-                            if(id) {
-                                this.props.store.dispatch(
-                                    mapSourceActions.modifyFeatureGeometry(map_source_name, id, geometry)
-                                );
-                            }
+                    this.drawTool.on('drawend', evt => {
+                        const querySourceName = util.getMapSourceName(layer.queryAs[0]);
+                        const querySource = this.props.mapSources[
+                            querySourceName
+                        ];
+                        let queryFeature = util.featureToJson(evt.feature);
+                        const mapProjection = this.map.getView().getProjection();
+
+                        if (querySource.config['buffer-point']) {
```

Comment:
@klassenjs and @brentfraser - Thoughts on this pattern for querying points/lines layers?

Reply:
Related to #392, #536, #601.  It's an OK fix.  But someone should do some research on the OGC protocols to see if there is something in the protocol to do a spatial query byt point geom.